pkg/shared/utils/utils.go

```diff
@@ -53,6 +58,39 @@ func RunCommand(cmd *exec.Cmd) ([]byte, error) {
 	return outb.Bytes(), nil
 }
 
+func RunCommandAndParseOutputLineByLine(cmd *exec.Cmd, pfn parserFn, resChan chan any, logger *logrus.Entry) error {
+	// Get a pipe to read from standard out
+	r, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("failed to get command pipe: %w", err)
+	}
+
+	scanner := bufio.NewScanner(r)
+	// Use the scanner to scan the output line by line and parse it
+	done := make(chan struct{})
+	go func() {
+		// Read line by line and process it
+		for scanner.Scan() {
+			line := scanner.Text()
+			resChan <- pfn(line, logger)
```

Comment:
I wouldn't add a channel here, the caller can handle their own result/error handling process in the closure they parse into the RunCommandAndParseOutputLineByLine function. For example in the Yara code:

```
detectedMalware := []common.DetectedMalware{}
parseErrs = []error{}
processFunc := func(ctx context.Context, line string) {
	malware, err := util.ParseYaraScanOutput(ctx, line)
    if err != nil {
    	errs = append(errs, err)
    	return
    }
    if malware != nil {
    	detectedMalware = append(detectedMalware, malware)
    }
}
err := RunCommandAndParseOutputLineByLine(cmd, processFunc)
if err != nil {
    return nil, fmt.Errorf("error running command: %w", err)
}
err := errors.Join(parseErrs...)
if err != nil {
   return nil, fmt.Errorf("error parsing lines: %w", err)
}
```